x-pack/plugins/cases/server/services/user_actions/index.ts

```diff
@@ -34,52 +29,44 @@ interface PostCaseUserActionArgs extends ClientArgs {
   actions: UserActionItem[];
 }
 
-export interface CaseUserActionServiceSetup {
-  getUserActions(
-    args: GetCaseUserActionArgs
-  ): Promise<SavedObjectsFindResponse<CaseUserActionAttributes>>;
-  postUserActions(
-    args: PostCaseUserActionArgs
-  ): Promise<SavedObjectsBulkResponse<CaseUserActionAttributes>>;
-}
-
 export class CaseUserActionService {
   constructor(private readonly log: Logger) {}
-  public setup = async (): Promise<CaseUserActionServiceSetup> => ({
-    getUserActions: async ({ client, caseId, subCaseId }: GetCaseUserActionArgs) => {
-      try {
-        const id = subCaseId ?? caseId;
-        const type = subCaseId ? SUB_CASE_SAVED_OBJECT : CASE_SAVED_OBJECT;
-        const caseUserActionInfo = await client.find({
-          type: CASE_USER_ACTION_SAVED_OBJECT,
-          fields: [],
-          hasReference: { type, id },
-          page: 1,
-          perPage: 1,
-        });
-        return await client.find({
-          type: CASE_USER_ACTION_SAVED_OBJECT,
-          hasReference: { type, id },
-          page: 1,
-          perPage: caseUserActionInfo.total,
-          sortField: 'action_at',
-          sortOrder: 'asc',
-        });
-      } catch (error) {
-        this.log.error(`Error on GET case user action case id: ${caseId}: ${error}`);
-        throw error;
-      }
-    },
-    postUserActions: async ({ client, actions }: PostCaseUserActionArgs) => {
-      try {
-        this.log.debug(`Attempting to POST a new case user action`);
-        return await client.bulkCreate(
-          actions.map((action) => ({ type: CASE_USER_ACTION_SAVED_OBJECT, ...action }))
-        );
-      } catch (error) {
-        this.log.error(`Error on POST a new case user action: ${error}`);
-        throw error;
-      }
-    },
-  });
+
+  public async getAll({ soClient, caseId, subCaseId }: GetCaseUserActionArgs) {
+    try {
+      const id = subCaseId ?? caseId;
+      const type = subCaseId ? SUB_CASE_SAVED_OBJECT : CASE_SAVED_OBJECT;
+      const caseUserActionInfo = await soClient.find<CaseUserActionAttributes>({
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+        fields: [],
+        hasReference: { type, id },
+        page: 1,
+        perPage: 1,
+      });
+
+      return await soClient.find<CaseUserActionAttributes>({
+        type: CASE_USER_ACTION_SAVED_OBJECT,
+        hasReference: { type, id },
+        page: 1,
+        perPage: caseUserActionInfo.total,
+        sortField: 'action_at',
+        sortOrder: 'asc',
+      });
```

Comment:
If I'm understanding this correctly, we are performing the first `find` in order to get the total count of user actions, and then running `find` again to get the full set of user actions (assuming it's under the default limit of 10,000).

If that's right, then I have two questions:
1) Can you just set `perPage` to `10,000` and make a single call? This already won't retrieve SOs over that count.
2) If not, can you use the new "point in time" helper to iterate over all user actions? I expect something like the following would work:
```suggestion
     const finder = soClient.createPointInTimeFinder<CaseUserActionAttributes>({
        type: CASE_USER_ACTION_SAVED_OBJECT,
        hasReference: { type, id },
        sortField: 'action_at',
        sortOrder: 'asc',
      });

      let result: Array<SavedObject<CaseUserActionAttributes>> = [];
      for await (const userActionSavedObject of finder.find()) {
        result = result.concat(userActionSavedObject.saved_objects);
      }
      return result;
```

Reply:
oh that's cool, I did not know about `createPointInTimeFinder`. We also want to refactor this code to work with `search_after` and lazy load all the data in case when you start scrolling. I just wanted to let you know.

Reply:
> Can you just set perPage to 10,000 and make a single call? This already won't retrieve SOs over that count.

Yeah this is probably the easiest change for now until we can support `search_after`. There are a couple of places where we have to do this same pattern of two calls. @legrego would you prefer that the changes go in this PR or could we do them in a separate one?

Reply:
I'm fine to have you do this in a followup if that's easier

Reply:
Added to this issue to track the work: https://github.com/elastic/kibana/issues/100804